File: src/main/java/com/gmail/borlandlp/minigamesdtools/localizer/LocalizerAPI.java
package com.gmail.borlandlp.minigamesdtools.localizer;

import org.bukkit.entity.Player;

import java.util.UUID;

/**
 * Компонент, хранящий языковые настройки игрока, а также отвечающий за хранение локализованного текста.
 */
public interface LocalizerAPI {
    /**
     * Устанавливает лвухзначное представление языка.
     * @param uuid
     */
    void setLanguage(UUID uuid, String langCode);

    /**
     * @param uuid
     * @return Двузначное представление языка. en, ru, de
     */
    String getLanguage(UUID uuid);

    String getLocalizedText(UUID uuid, String textId);
    String getLocalizedText(String langCode, String textId);
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/localizer/LocalizerAPI.kt
package com.gmail.borlandlp.minigamesdtools.localizer

import java.util.*

/**
 * Компонент, хранящий языковые настройки игрока, а также отвечающий за хранение локализованного текста.
 */
interface LocalizerAPI {
    /**
     * Устанавливает лвухзначное представление языка.
     * @param uuid
     */
    fun setLanguage(uuid: UUID, langCode: String)

    /**
     * @param uuid
     * @return Двузначное представление языка. en, ru, de
     */
    fun getLanguage(uuid: UUID): String

    fun getLocalizedText(uuid: UUID, textId: String): String?
    fun getLocalizedText(langCode: String, textId: String): String?
}